File: 김순석/프로그래머스/pro_스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for tree in skill_trees:
        tmp = ''
        for i in range(len(tree)):
            if tree[i] in skill:
                tmp += tree[i]
        for i in range(len(tmp)):
            if skill[i] != tmp[i]:
                break
        else:
            answer += 1
    return answer

File: AHYEON/13.programmers_lv2/pro_스킬트리.swift
import Foundation

func solution(_ skill:String, _ skill_trees:[String]) -> Int {
    var cnt = 0
    for s in skill_trees {
        var currentSkill = ""
        for c in s {
            if skill.contains(c) {
                currentSkill += c.description
            }
        }
        
        if skill.hasPrefix(currentSkill) {
            cnt += 1
        }
    }
    return cnt
}